app/services/PaypalService.scala

```diff
@@ -59,27 +87,69 @@ class PaypalService(config: PaypalApiConfig) {
       val links = createdPayment.getLinks.asScala
       val approvalLink = links.find(_.getRel.equalsIgnoreCase("approval_url"))
       approvalLink.map(l => Right(l.getHref)).getOrElse(Left("No approval link returned from paypal"))
-    }
-    catch {
+    } catch {
       case e: PayPalRESTException => Left(e.getMessage)
     }
   }
 
-  def executePayment(paymentId: String, token: String, payerId: String): Either[String, Unit] = {
+  def executePayment(paymentId: String, payerId: String): Either[String, String] = {
     val payment = new Payment().setId(paymentId)
     val paymentExecution = new PaymentExecution().setPayerId(payerId)
     try {
       val createdPayment = payment.execute(apiContext, paymentExecution)
       if (createdPayment.getState.toUpperCase != "APPROVED") {
         Left(s"payment returned with state: ${createdPayment.getState}")
+      } else {
+        Payment.get(apiContext, paymentId)
+        Right(createdPayment.getId)
       }
-      else
-        Right(Unit)
     } catch {
       case e: PayPalRESTException => Left(e.getMessage)
     }
   }
 
+  def storeMetaData(
```

Comment:
I think it's completely specific. The best we could would be to extract it to a trait, but even then, I doubt the signature would match with what stripe has